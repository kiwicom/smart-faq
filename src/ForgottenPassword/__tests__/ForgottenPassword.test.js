// @flow
import * as React from 'react';
import { shallow } from 'enzyme';

import ForgottenPasword from '../index';

jest.mock('../../mutations/ResetPassword');

describe('ForgottenPassword', () => {
  const props = {
    history: {
      push: jest.fn(),
    },
  };
  const e = {
    preventDefault: jest.fn(),
  };

  const wrapper = shallow(<ForgottenPasword {...props} />);
  it('should change state', () => {
    const ForgottenPasswordIns = wrapper.instance();
    const state = ForgottenPasswordIns.state;
    expect(state).toEqual({ email: '' });

    ForgottenPasswordIns.handleChangeEmail({
      target: {
        value: 'email@kiwi.com',
      },
    });

    expect(ForgottenPasswordIns.state).toEqual({ email: 'email@kiwi.com' });
  });

  it('should handle email submission', async () => {
    const ForgottenPasswordIns = wrapper.instance();
    await ForgottenPasswordIns.handleSubmitEmail(e);

    expect(e.preventDefault.mock.calls).toHaveLength(1);
    expect(ForgottenPasswordIns.props.history.push.mock.calls[0][0]).toEqual({
      pathname: '/check-recovery-link',
      state: { email: 'email@kiwi.com' },
    });
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
