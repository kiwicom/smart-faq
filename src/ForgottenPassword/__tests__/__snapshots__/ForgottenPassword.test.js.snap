// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgottenPassword should match snapshot 1`] = `
<div
  className="jsx-2562091625 ForgottenPassword"
>
  <CloseButton />
  <withRouter(BackButton)
    prevScreen="kiwi-login"
  />
  <div
    className="jsx-2562091625 picture"
  >
    <Illustration
      name="Help"
      size="medium"
    />
  </div>
  <div
    className="jsx-2562091625 main"
  >
    <p
      className="jsx-2562091625 title"
    >
      <Text
        html={false}
        t="smartfaq.forgotten_password.title"
        values={Object {}}
      />
    </p>
    <Text
      type="secondary"
    >
      <Text
        html={false}
        t="smartfaq.forgotten_password.subtitle"
        values={Object {}}
      />
    </Text>
  </div>
  <form
    className="jsx-2562091625"
    onSubmit={[Function]}
  >
    <label
      className="jsx-2562091625"
      htmlFor="email"
    >
      <Text>
        <Text
          html={false}
          t="smartfaq.forgotten_password.email_label"
          values={Object {}}
        />
      </Text>
      <div
        className="jsx-2562091625 input"
      >
        <InputText
          onChange={[Function]}
          placeholder="e.g. your@email.com"
          required={true}
          type="email"
          value=""
        />
      </div>
    </label>
    <button
      className="jsx-2562091625 send"
    >
      <Text
        html={false}
        t="smartfaq.forgotten_password.send_button"
        values={Object {}}
      />
    </button>
  </form>
  <JSXStyle
    css=".ForgottenPassword.jsx-2562091625{width:480px;padding-top:128px;margin:auto;}div.picture.jsx-2562091625{text-align:center;margin-bottom:68px;}div.main.jsx-2562091625{margin-left:40px;margin-right:40px;}p.title.jsx-2562091625{color:#171b1e;font-size:28px;font-weight:bold;margin-top:0;margin-bottom:8px;line-height:1.2;}button.send.jsx-2562091625{width:83px;background-color:#00a991;border:0;font-size:14px;font-weight:bold;line-height:1.43;color:#ffffff;}label.jsx-2562091625{display:inline-block;width:300px;height:44px;}form.jsx-2562091625{margin:32.8px 40px 0px 40px;}div.input.jsx-2562091625{display:inline-block;width:100%;height:44px;margin-top:4px;}form.jsx-2562091625 button.jsx-2562091625{width:83px;height:44px;border-radius:3px;background-color:#00a991;font-size:14px;color:#ffffff;cursor:pointer;font-weight:bold;line-height:1.43;text-align:center;border:none;margin-left:8px;display:inline-block;}@media only screen and (max-width:480px){.ForgottenPassword.jsx-2562091625{width:100%;padding-top:64px;}div.picture.jsx-2562091625{display:none;}p.title.jsx-2562091625{font-size:22px;text-align:left;}div.main.jsx-2562091625{margin-left:16px;margin-right:16px;}form.jsx-2562091625{margin:33px 16px 0px;}label.jsx-2562091625{display:block;width:100%;height:44px;margin-bottom:33px;}form.jsx-2562091625 button.jsx-2562091625{width:100%;border-radius:3px;margin-bottom:0px;margin-left:0px;display:block;width:100%;}button.send.jsx-2562091625{width:288px;height:44px;}}@media only screen and (max-height:480px) and (orientation:landscape){.ForgottenPassword.jsx-2562091625{width:288px;padding-top:20px;}div.picture.jsx-2562091625{display:none;}p.title.jsx-2562091625{font-size:22px;text-align:center;}div.main.jsx-2562091625{margin:0;}div.input.jsx-2562091625{width:288px;}form.jsx-2562091625{margin:33px 0 0;}form.jsx-2562091625 button.jsx-2562091625{width:100%;margin:28px 0 0;}}"
    styleId="2562091625"
  />
</div>
`;
